### PR TITLE
DRILL-4733: max(dir0) reading more columns than necessary

### PR DESCRIPTION
**Pull request by arina-ielchiieva**

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/easy/EasyFormatPlugin.java [hunk 1]
```diff
@@ -126,8 +127,12 @@ CloseableRecordBatch getReaderBatch(FragmentContext context, EasySubScan scan) t
     final ImplicitColumnExplorer columnExplorer = new ImplicitColumnExplorer(context, scan.getColumns());
 
     if (!columnExplorer.isSelectAllColumns()) {
+      // We must make sure to pass a table column (not to be confused with implicit column) to the underlying record reader.
+      List<SchemaPath> tableColumns =
```
Comment (amansinha100) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/easy/EasyFormatPlugin.java]: I haven't looked at the original patch for implicit columns but I am not sure why this fix is in the EasyFormatPlugin when the test is against Parquet files ?

File: exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java [hunk 2]
```diff
@@ -110,4 +111,20 @@ public void testImplicitColumnsForParquet() throws Exception {
         .go();
   }
 
+  @Test // DRILL-4733
+  public void testMultilevelParquetWithSchemaChange() throws Exception {
+    try {
+      test("alter session set `planner.enable_decimal_data_type` = true");
```
Comment (amansinha100) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java]: Why is decimal type relevant for this particular test ?

Comment (arina-ielchiieva) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java]: When I run this query without decimal data type enabled, drill tell me to turn it on. Probably it's connected with data inside the dataset.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/easy/EasyFormatPlugin.java (hunk 1 above)
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/easy/EasyFormatPlugin.java]: In original PR  I have created helper class which contained common logic for parquet and test format plugins. Somehow I missed that this part is unique for text format plugin, and should NOT be used in parquet one. That's why I have removed it from ImplicitColumnExplorer and added to EasyFormatPlugin.

File: exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java (hunk 2 above)
Comment (rchallapalli) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java]: One of the parquet files in the data set contain a column which is double. But I do not understand why drill requires us to enable the decimal type;

File: exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java [hunk 3]
```diff
@@ -110,4 +111,20 @@ public void testImplicitColumnsForParquet() throws Exception {
...
+      testBuilder()
+          .sqlQuery(String.format("select max(dir0) as max_dir from dfs_test.`%s/src/test/resources/multilevel/parquetWithSchemaChange`",
+              TestTools.getWorkingPath()))
+          .unOrdered()
+          .baselineColumns("max_dir")
+          .baselineValues("voter50.parquet")
```
Comment (jinfengni) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java]: Why do you put baselineValue in a parquet, in stead of putting it in the testcase directly? Tthe query seems to return one single value.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/easy/EasyFormatPlugin.java (hunk 1 above)
Comment (amansinha100) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/easy/EasyFormatPlugin.java]: Ok, I see.   Although this patch resolves this issue, I am thinking that without doing a performance test it is not feasible to see the performance impact of the overall implicit columns support.  It is a nice feature to have but I think we can give it a little more time to go through functional and perf tests.

File: exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java (hunk 3 above)
Comment (arina-ielchiieva) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java]: @jinfengni 
I guess confusion in here is that `voter50.parquet` is folder name. If would be clearer I can rename folder and files in it (currently files have names 0_0_0.parquet).

Comment (jinfengni) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java]: Got it. "voter50.parquet" is actually a folder name. I thought it's a parquet file.

Comment (amansinha100) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java]: It does seem confusing to have the .parquet extension to the folder name.  Could you change it and update the pull request ? 

Perf tests look good so far.  

+1 on the patch.

Comment (arina-ielchiieva) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/TestImplicitFileColumns.java]: Done.

**Pull request closed without merging**